dynawo/sources/ModelicaCompiler/Scripts_OMC_1_13_2/readerOMC.py

```diff
@@ -1236,95 +1245,187 @@ def read_05evt_c_file(self):
-                                    structure1.list_elements.append(s)
-                                    structure1=s
-                                else:
-                                    structure1 = self.map_struct_name_struct_obj[struct_name1]
-
-                        terminal = Element(False,name_terminal,len(self.list_elements))
-                        self.list_elements.append(terminal)
-                        structure1.list_elements.append(terminal)
-
-                        #Find vars that are "flow"
-                        if type_connector == "Flow" :  self.list_flow_vars.append(name)
+                        raise UnknownElementsElement(elements_terminal_element.tag)
+            else:
+               raise UnknownModelElement(model_element.tag)
+
+    ##
+    # parse struct etree element
+    # @param self : object pointer
+    # @param list_struct : structs list
+    # @param struct_elements : struct etree element
+    # @return
+    def parse_struct(self, list_struct, struct_elements):
+        number_of_struct_names = 0
+        for struct_subelem in struct_elements:
+            if struct_subelem.tag == "name":
+                number_of_struct_names += 1
+                if number_of_struct_names >= 2:
+                    raise StructHasMoreThanOneName
+                struct_name = struct_subelem.text
+            elif struct_subelem.tag == "subnodes":
+                for subnodes_element in struct_subelem:
+                    if subnodes_element.tag == "terminal":
+                        self.parse_struct_terminal(list_struct, subnodes_element, struct_name)
+                    else:
+                        raise UnknownSubnodesElement(subnodes_element.tag)
+            else:
+                raise UnknownStructElement(struct_subelem.tag)
+
+    ##
+    # parse terminal etree element contained in struct etree element
+    # @param self : object pointer
+    # @param list_struct : structs list
+    # @param struct_terminal_element : terminal etree element contained in struct etree element
+    # @param struct_name : struct name
+    # @return
+    def parse_struct_terminal(self, list_struct, struct_terminal_element, struct_name):
+        number_of_struct_terminal_names = 0
+        number_of_struct_terminal_connectors = 0
+        for struct_terminal_subelem in struct_terminal_element:
+            if struct_terminal_subelem.tag == "name":
+                number_of_struct_terminal_names += 1
+                if number_of_struct_terminal_names >= 2:
+                    raise TerminalHasMoreThanOneName
+                name_terminal = struct_terminal_subelem.text
+            elif struct_terminal_subelem.tag == "connector":
+                number_of_struct_terminal_connectors += 1
+                if number_of_struct_terminal_connectors >= 2:
+                    raise TerminalHasMoreThanOneConnector
+                type_connector = struct_terminal_subelem.attrib.get('type', '')
```

Comment:
If connector XML element has no type attribute, type connector would be an empty string. But this modification is only necessary to make the unit tests pass so I'm not sure it's necessary.